common/display/displayplanemanager.cpp

```diff
@@ -87,8 +88,15 @@ bool DisplayPlaneManager::ValidateLayers(
   // primary layer cannot be scanned out directly.
   bool prefer_seperate_plane = primary_layer->PreferSeparatePlane();
   bool force_gpu = (pending_modeset && layers.size() > 1) || disable_overlay;
+  bool force_va = false;
 
-  if (force_gpu || FallbacktoGPU(current_plane, primary_layer, commit_planes)) {
+  // If request video effect we have to use VA to process the video layer
+  if (request_video_effect && prefer_seperate_plane) {
+    force_va = true;
+  }
+
+  if (force_gpu || FallbacktoGPU(current_plane, primary_layer, commit_planes) ||
+      force_va) {
```

Comment:
I think everytime we have these chacks: if (primary_layer->IsVideoLayer()) you can just also check if force_va and than set render_layers = true and also call ForceGPURendering() for plane state. You want to create a helper function and call it from all places which have IsVideoLayer checks (Move calls like SetVideoPlane to that function).

Reply:
I am not quite understand... Isn't primary_layer->IsVideoLayer check too late?

Reply:
No the check should be fine. What we want to know really is that even if video buffer can be scanned out by plane, we need to handle the post processing and not scan it out directly.  Keeps all Video/Camera checks together.